odoo_sh/advanced/submodules.rst

```diff
@@ -0,0 +1,104 @@
+:banner: banners/odoo-sh.jpg
+
+.. _odoosh-advanced-submodules:
+
+==================================
+Submodules
+==================================
+
+Overview
+========
+
+A `Git submodule <https://git-scm.com/book/en/v2/Git-Tools-Submodules>`_ allows you to integrate other Git projects
+into your code, without the need to copy-paste all their code.
+
+Indeed, your custom modules can depend on modules from other repositories.
+Regarding Odoo, this feature allows you to add modules from other Git repositories into the branches of your repository.
+Adding these dependencies within your branch through submodules makes easier the deployment of your code and servers,
+as you can clone the repositories added as submodules in the same time you clone your own repository.
+
+Besides, you can choose the branch of the repository added as submodule
+and you have the control of the revision you want.
+It's up to you to decide wether you want to pin the submodule to a specific revision and when you want to update
+to a newer revision.
+
+In Odoo.sh, the submodules gives you the possibility to use and depends on modules available in other repositories.
+The platform will detect automatically that you added modules through submodules within your branches,
+and add them within your addons path so you can install them in your databases.
```

Comment:
The platform will detect that you added modules through submodules in your branches and add them to your addons path automatically so you can install them in your databases.